### PR TITLE
Allow to connect to Azure File Storage with Azure Managed Identity

### PR DESCRIPTION
**Pull request by fabiomaistro**
## Description
Allow to connect to Azure File Storage with Azure Managed Identity

## Motivation and Context
Improve usability

### CODE REVIEW

File: src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs [hunk 1]
```diff
@@ -1,13 +1,76 @@
+using OneBeyond.Studio.FileStorage.Azure.Exceptions;
 using System;
+using System.Text.RegularExpressions;
 
 namespace OneBeyond.Studio.FileStorage.Azure.Options;
 
 public abstract record AzureBaseStorageOptions
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs:7]: Fabio, those options, are those the ones we retrieve from the configuration file (appsettings.json)? My understanding was we apply same rules to those classes as to DTOs, so just simplest serialiable data containers with no logic inside.

**Review by andriikaplanovskyi — APPROVED** [review]

File: src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs [hunk 2]
```diff
@@ -1,13 +1,76 @@
...
+
+    /// <summary>
+    /// The name of the Azure Storage Account to use for Azure Identity authentication.
+    /// NOTE: If specified, a connection string must not be provided.
+    /// </summary>
+    public string? AccountName { get; init; }
+
+    /// <summary>
+    /// The connection string to use for authentication.
+    /// NOTE: if specified, a resource name must not be provided.
```
Comment (danieltoth55) [inline src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs:17]: account name*

**Review by danieltoth55 — APPROVED** [review]

File: src/OneBeyond.Studio.FileStorage.Azure/OneBeyond.Studio.FileStorage.Azure.csproj [hunk 3]
```diff
@@ -11,7 +11,9 @@
   </PropertyGroup>
 
   <ItemGroup>
+	<PackageReference Include="Azure.Identity" Version="1.9.0" />
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.FileStorage.Azure/OneBeyond.Studio.FileStorage.Azure.csproj:14]: BTW, problems wit indentation here

File: src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs [hunk 4]
```diff
@@ -2,12 +2,27 @@
 
 namespace OneBeyond.Studio.FileStorage.Azure.Options;
 
-public abstract record AzureBaseStorageOptions
+public abstract partial record AzureBaseStorageOptions
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.FileStorage.Azure/Options/AzureBaseStorageOptions.cs:5]: @fabiomaistro when you use your  AzureStorageOptionsValidator class, does this `AzureBaseStorageOptions` class have to be partial?

File: src/OneBeyond.Studio.FileStorage.Azure/Options/Validators/AzureStorageOptionsValidator.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
+﻿using System.Text.RegularExpressions;
+using Microsoft.Extensions.Options;
+
+namespace OneBeyond.Studio.FileStorage.Azure.Options.Validators
+{
+    public partial class AzureStorageOptionsValidator<T> : IValidateOptions<T>
```
Comment (andriikaplanovskyi) [inline src/OneBeyond.Studio.FileStorage.Azure/Options/Validators/AzureStorageOptionsValidator.cs:6]: Fabio, checking your example, did not understand why this class is also partial

**Review by andriikaplanovskyi — APPROVED** [review]

**Pull request merged**